src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -931,19 +925,35 @@ abstract class ManagedStrategy(
             val count = currentThreadCodeLocationVisitCountMap.getOrDefault(codeLocation, 0) + 1
             currentThreadCodeLocationVisitCountMap[codeLocation] = count
             currentThreadCodeLocationsHistory += codeLocation
+            val detectedFirstTime = count > hangingDetectionThreshold
             val detectedEarly = loopTrackingCursor.isInCycle
             // Check whether the count exceeds the maximum number of repetitions for loop/hang detection.
-            val detectedFirstTime = count > hangingDetectionThreshold
             if (detectedFirstTime && !detectedEarly) {
                 registerCycle()
+                // Enormous operations count considered as total spin lock
+                if (totalExecutionsCount > ManagedCTestConfiguration.LIVELOCK_EVENTS_THRESHOLD) {
```

Comment:
By the way, do we now shorten the execution when detecting a livelock? If not, please raise an issue.

Reply:
Yes, we do that, please consider: `org.jetbrains.kotlinx.lincheck_test.representation.SpinlockEventsCutMiddleLengthTest`,
`org.jetbrains.kotlinx.lincheck_test.representation.SpinlockEventsCutLongCycleActionsTest`, etc.

Reply:
In `SpinlockEventsCutLongCycleActionsTest`, how are we sure that switching back to the previous thread would not prevent the live lock? Additionally to that, I see a big scenario in the output.

Reply:
The scenario is big due to `.minimizeFailedScenario(false)` configuration. In the next PR related to recursive spin locks support and enhanced cycle period detection, I can turn it off, and a much less scenario will be found (Anyway, the output will be changed because arrows will be added).

Reply:
Sure, let's do that.